code/processes/chainedtp.q

```diff
@@ -195,13 +195,13 @@ upd:.ctp.upd;
 /- pubsub must be initialised sooner to enable tickerplant replay publishing to work
 .ps.initialise[];                                                                   
 
+.servers.startup[]; 
 /- check if the tickerplant has connected, blocks the process until a connection is established
 .ctp.initialise[];
-while[.ctp.notpconnected[];
-  /- while not connected make the process sleep for X seconds and then run the subscribe function again
-  .os.sleep[.ctp.tpconnsleep];
-  /- start chained tickerplant
-  .ctp.initialise[]];
+
+//check if tickerplant is available and if not exit with error
+.servers.startupdepcycles[.ctp.requiredprocs;.ctp.tpconnsleep;.ctp.tpcheckcycles]; 
+.ctp.initialise[]; 
```

Comment:
I think you should back out all the changes from this file, except the spacing changes.  

The CTP subscribes based on TP name rather than type, so the new logic doesn't quite fit here.